src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointAlignment.java

```diff
@@ -0,0 +1,170 @@
+     * Assumes {@code region1} has a lower {@link AlignmentRegion#startInAssembledContig} than {@code region2}.
+     */
+    public BreakpointAlignment(final String contigId, final AlignmentRegion region1, final AlignmentRegion region2, final String insertedSequence, final String homology, final List<String> insertionMappings) {
+        this.contigId = contigId;
+        this.region1 = region1;
+        this.region2 = region2;
+        this.insertedSequence = insertedSequence;
+        this.homology = homology;
+        this.insertionMappings = insertionMappings;
+    }
+
+    @Override
+    public String toString() {
+        return contigId +
+                "\t" +
+                region1.toString() +
+                "\t" +
+                region2.toString() +
+                "\t" +
+                ("".equals(insertedSequence) ? NO_SEQUENCE : insertedSequence) +
+                "\t" +
+                ("".equals(homology) ? NO_SEQUENCE : homology);
+    }
+
+    /**
+     *  Parses a tab-delimited assembled breakpoint line into an BreakpointAlignment object. Fields should be in the same order as that produced by toString():
+     *
+     *  contigId
+     *  alignmentRegion1.toString()
+     *  alignmentRegion2.toString()
+     *  insertedSequence
+     *  homology
+     *
+     */
+    public static BreakpointAlignment fromString(String assembledBreakpointLine) {
+        final String[] fields = assembledBreakpointLine.split("\t");
+        return fromFields(fields);
+    }
+
+    private static BreakpointAlignment fromFields(final String[] fields) {
+        try {
+            final String contigId = fields[0].replaceFirst("^>","");
+            final String[] alignmentRegion1Fields = Arrays.copyOfRange(fields, 1, 10);
+            final AlignmentRegion alignmentRegion1 = AlignmentRegion.fromString(alignmentRegion1Fields);
+            final String[] alignmentRegion2Fields = Arrays.copyOfRange(fields, 10, 19);
+            final AlignmentRegion alignmentRegion2 = AlignmentRegion.fromString(alignmentRegion2Fields);
+            final String insertedSequence = fields[19].equals(NO_SEQUENCE) ? "" : fields[19];
+            final String homology = fields[20].equals(NO_SEQUENCE) ? "" : fields[20];
+            final List<String> insertionMappings = Arrays.asList(fields[21].split(";"));
+            return new BreakpointAlignment(contigId, alignmentRegion1, alignmentRegion2, insertedSequence, homology, insertionMappings);
+        } catch (final NumberFormatException nfe) {
+            throw new GATKException(Arrays.toString(fields), nfe);
+        }
+
+    }
+
+    private AlignmentRegion getLeftAlignmentRegion() {
+        if (region1.referenceInterval.getStart() < region2.referenceInterval.getStart()) {
+            return region1;
+        } else {
+            return region2;
+        }
+    }
+
+    @VisibleForTesting
+    SimpleInterval getLeftAlignedLeftBreakpointOnReference() {
```

Comment:
As we discussed, another possible change that easier for me to understand but not necessarily true for others:
the private method `getLeftAlignmentRegion()` is picking the AR (AlignmentRegion) with the lower reference-coordinate (starting position really, we could dream up a case that one is completely contained in the other but that's impossible for a good inversion contig, for now). So the three methods: `getLeftAlignmentRegion()`, `getLeftAlignedLeftBreakpointOnReference()`, `getLeftAlignedRightBreakpointOnReference()`, could be merged into a single method, something like:

```java
    /**
     * Computes the 5'- and 3'- end breakpoint locations, 
     * where 5'-end BP is left aligned on reference and 3'-end BP is right aligned when homology around breakpoints is present.
     */
    @VisibleForTesting
    Tuple2<SimpleInterval, SimpleInterval> inferBreakpointLocWithParsimony(){

        String fiveContig, threeContig;
        int fivePosition, threePosition;

        if (region1.referenceInterval.getStart() < region2.referenceInterval.getStart()) {
            fivePosition = region1.forwardStrand ? region1.referenceInterval.getEnd() - homology.length() : region1.referenceInterval.getStart();
            threePosition = region2.forwardStrand ? region2.referenceInterval.getStart() + homology.length() : region2.referenceInterval.getEnd();
            fiveContig = region1.referenceInterval.getContig();
            threeContig = region2.referenceInterval.getContig();
        } else {
            fivePosition = region2.forwardStrand ? region2.referenceInterval.getStart() : region2.referenceInterval.getEnd() - homology.length();
            threePosition = region1.forwardStrand ? region1.referenceInterval.getEnd() : region1.referenceInterval.getStart() + homology.length() ;
            fiveContig = region2.referenceInterval.getContig();
            threeContig = region1.referenceInterval.getContig();
        }

        return new Tuple2<>( new SimpleInterval(fiveContig, fivePosition, fivePosition),
                             new SimpleInterval(threeContig, threePosition, threePosition) );
    }
``` 

Reason for naming this function this way (Soohee helped me wit this): `BreakpointAlignment`, for an inversion, is either 5-to-3 or 3-to-5. (Another language system that I personally also use is 5-5 adjacency and 3-3 adjacency, to be inline with the "novel adjacency" language system.) The logic here is (IMO) to give the homologous sequence to 
1. the inferred 3'-end (therefore 5'-end BP is left-aligned) in the case of 5-to-3 type (or the 5-5 adjacency);
2. the inferred 5'-end (therefore 3'-end BP is right-aligned) in the case of 3-to-5 type (or the 3-3 adjacency);